### PR TITLE
K8SPSMDB-400: Simplify secret change logic

### PR DESCRIPTION
**Pull request by blez**
[K8SPSMDB-381](https://jira.percona.com/browse/K8SPSMDB-381) [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

In current implementation operator saves hashes of secrets to annotations. This is done for restarting pods in case of secret changing. Pods must be restarted to make sure `pbm` and `pmm` containers reread ENV vars (with credentials).

This PR simplifies it a bit. Operator does not restart whole pod, but just kills `pbm` and `pmm` containers. `mongod` container proceeds to work.

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| self-healing | passed |
| scheduled-backup | passed |
| security-context | passed |
| arbiter | passed |
| demand-backup-sharded | failed |
| operator-self-healing | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/5b7a3aa55bd9514128458389b7b067f3dba67b7b
image: `perconalab/percona-server-mongodb-operator:PR-599-5b7a3aa5`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| one-pod | passed |
| security-context | passed |
| monitoring-2-0 | failed |
| rs-shard-migration | passed |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/5b7a3aa55bd9514128458389b7b067f3dba67b7b
image: `perconalab/percona-server-mongodb-operator:PR-599-5b7a3aa5`

File: pkg/controller/perconaservermongodb/users.go [hunk 1]
```diff
@@ -60,52 +59,105 @@ func (r *ReconcilePerconaServerMongoDB) reconcileUsers(cr *api.PerconaServerMong
...
+	if restartMongos {
+		list, err := r.getMongosPods(cr)
+		if err != nil {
+			return errors.Wrap(err, "failed to get mongos pods")
+		}
+
+		for _, name := range containers {
+			err = r.killcontainer(list.Items, name)
+			if err != nil {
+				return errors.Wrap(err, "failed to kill pmm-client container")
```
Comment (delgod) [inline pkg/controller/perconaservermongodb/users.go:103]:
```suggestion
				return errors.Wrap(err, "failed to kill %s container", container)
```

File: pkg/controller/perconaservermongodb/users.go [hunk 2]
```diff
@@ -60,52 +59,105 @@ func (r *ReconcilePerconaServerMongoDB) reconcileUsers(cr *api.PerconaServerMong
...
+		return nil
 	}
 
 	restartSfs, restartMongos, err := r.updateSysUsers(cr, &sysUsersSecretObj, &internalSysSecretObj, repls)
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "manage sys users")
+		return errors.Wrap(err, "manage sys users")
+	}
+
+	containers := []string{"pmm-client", "backup-agent"}
```
Comment (delgod) [inline pkg/controller/perconaservermongodb/users.go:92]: why we need to restart pmm if only the backup password was changed and vise versa

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/controller/perconaservermongodb/users.go (hunk 1 above)
Comment (blez) [reply pkg/controller/perconaservermongodb/users.go]: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/206707ae3e46190cb6a6979c5ff0837156da3c3b

File: pkg/psmdb/mongos.go [hunk 3]
```diff
@@ -128,6 +128,11 @@ func mongosContainer(cr *api.PerconaServerMongoDB) (corev1.Container, error) {
 			MountPath: sslInternalDir,
 			ReadOnly:  true,
 		},
+		{
```
[the remark this answers is not in the document]
Comment (blez) [reply pkg/psmdb/mongos.go]: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/206707ae3e46190cb6a6979c5ff0837156da3c3b

File: healthcheck/tools/db/config.go [hunk 4]
```diff
@@ -82,6 +83,8 @@ func NewConfig(app *kingpin.Application, envUser string, envPassword string) (*C
 
 	pwdFile := "/etc/users-secret/MONGODB_CLUSTER_MONITOR_PASSWORD"
 	if _, err := os.Stat(pwdFile); err == nil {
+		log.Info("reading password from secret")
```
[the remark this answers is not in the document]
Comment (blez) [reply healthcheck/tools/db/config.go]: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/206707ae3e46190cb6a6979c5ff0837156da3c3b

File: pkg/psmdb/mongos.go [hunk 5]
```diff
@@ -285,6 +293,14 @@ func volumes(cr *api.PerconaServerMongoDB) []corev1.Volume {
 				EmptyDir: &corev1.EmptyDirVolumeSource{},
 			},
 		},
+		{
+			Name: "users-secret-file",
```
Comment (delgod) [inline pkg/psmdb/mongos.go:297]: I think it should be added only if cr version -> 1.8.0
to avoid mongos restart during operator upgrade

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/controller/perconaservermongodb/psmdb_controller.go [hunk 6]
```diff
@@ -238,9 +238,8 @@ func (r *ReconcilePerconaServerMongoDB) Reconcile(request reconcile.Request) (re
 		repls = append([]*api.ReplsetSpec{cr.Spec.Sharding.ConfigsvrReplSet}, repls...)
 	}
 
-	var sfsTemplateAnnotations, mongosTemplateAnnotations map[string]string
```
[the remark this answers is not in the document]
Comment (blez) [reply pkg/controller/perconaservermongodb/psmdb_controller.go:241]: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/485b4762b471dbb21be881fde6fb398a197fdeb4

File: pkg/psmdb/mongos.go (hunk 5 above)
Comment (blez) [reply pkg/psmdb/mongos.go:300]: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/c0bf393699e70c222aee61244a057c5c7b4507aa

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | failed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| arbiter | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | failed |
| service-per-pod | passed |
| upgrade | passed |
| liveness | passed |
| upgrade-sharded | failed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/226ffc88b4b5e8bd56b5ce08dfdee0b2535c25f4
image: `perconalab/percona-server-mongodb-operator:PR-599-226ffc88`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| self-healing | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| arbiter | passed |
| operator-self-healing | passed |
| scheduled-backup | failed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| demand-backup-sharded | failed |
| users | passed |
| upgrade | failed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/206707ae3e46190cb6a6979c5ff0837156da3c3b
image: `perconalab/percona-server-mongodb-operator:PR-599-206707ae`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| arbiter | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | failed |
| service-per-pod | passed |
| upgrade | passed |
| liveness | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/485b4762b471dbb21be881fde6fb398a197fdeb4
image: `perconalab/percona-server-mongodb-operator:PR-599-485b4762`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | failed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| arbiter | passed |
| security-context | passed |
| demand-backup-sharded | failed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| upgrade | passed |
| liveness | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/78a5cc20d34b7836d8624974209a037470b407a8
image: `perconalab/percona-server-mongodb-operator:PR-599-78a5cc20`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | failed |
| self-healing | passed |
| limits | passed |
| demand-backup | passed |
| scaling | passed |
| security-context | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| arbiter | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | failed |
| service-per-pod | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | failed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/dc8a73a32265ffe830dba09cd9215685f28fc3af
image: `perconalab/percona-server-mongodb-operator:PR-599-dc8a73a3`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| self-healing | passed |
| limits | passed |
| monitoring-2-0 | failed |
| scaling | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| arbiter | passed |
| demand-backup-sharded | failed |
| service-per-pod | passed |
| upgrade | passed |
| liveness | passed |
| upgrade-sharded | passed |
| smart-update | passed |
| version-service | passed |
| security-context | passed |
| rs-shard-migration | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/ddaf725ba730e0f644db70492abbbf3689caeffc
image: `perconalab/percona-server-mongodb-operator:PR-599-ddaf725b`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | failed |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/ddaf725ba730e0f644db70492abbbf3689caeffc
image: `perconalab/percona-server-mongodb-operator:PR-599-ddaf725b`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| limits | passed |
| self-healing | passed |
| scaling | passed |
| operator-self-healing | passed |
| security-context | passed |
| one-pod | passed |
| rs-shard-migration | passed |
| monitoring-2-0 | failed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/ddaf725ba730e0f644db70492abbbf3689caeffc
image: `perconalab/percona-server-mongodb-operator:PR-599-ddaf725b`

**Review by delgod — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| self-healing | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| smart-update | passed |
| demand-backup-sharded | passed |
| version-service | passed |
| upgrade | failed |
| users | passed |
| data-sharded | passed |
| upgrade-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/050c15f41a411507e166e68d59cc94e374865490
image: `perconalab/percona-server-mongodb-operator:PR-599-050c15f4`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| one-pod | passed |
| self-healing | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| scheduled-backup | passed |
| liveness | passed |
| demand-backup-sharded | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| upgrade | failed |
| data-sharded | passed |
| upgrade-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/04612dbb4e332529b2491dfdfa117a2107428cbc
image: `perconalab/percona-server-mongodb-operator:PR-599-04612dbb`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/1deb37237c15b13eb86b538ddede669fb8c92c12
image: `perconalab/percona-server-mongodb-operator:PR-599-1deb3723`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| operator-self-healing | passed |
| liveness | passed |
| smart-update | passed |
| upgrade | failed |
| version-service | passed |
| upgrade-sharded | failed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/1deb37237c15b13eb86b538ddede669fb8c92c12
image: `perconalab/percona-server-mongodb-operator:PR-599-1deb3723`

File: pkg/apis/psmdb/v1/psmdb_types.go [hunk 7]
```diff
@@ -549,6 +549,21 @@ func (cr *PerconaServerMongoDB) CompareVersion(version string) int {
 	return cr.Version().Compare(v.Must(v.NewVersion(version)))
 }
 
+func UserSecretNameInternal(cr *PerconaServerMongoDB) string {
+	internalPrefix := "internal-"
```
[the remark this answers is not in the document]
Comment (blez) [reply pkg/apis/psmdb/v1/psmdb_types.go:553]: It's used only here so there is no need to move it somewhere

[the remark this answers is not in the document]
Comment (qJkee) [reply pkg/apis/psmdb/v1/psmdb_types.go:553]: but we create it on every function entry.
Could save some performance in the future

[the remark this answers is not in the document]
Comment (qJkee) [reply pkg/apis/psmdb/v1/psmdb_types.go:553]: btw, we use this `internal-` prefix in more places, which could be refactored in the future

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| self-healing | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| operator-self-healing | passed |
| arbiter | passed |
| rs-shard-migration | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| demand-backup-sharded | passed |
| version-service | passed |
| users | passed |
| upgrade | failed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/9dabe4f352974f7f360ca6682b3c3f589de575c3
image: `perconalab/percona-server-mongodb-operator:PR-599-9dabe4f3`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| one-pod | passed |
| rs-shard-migration | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/9dabe4f352974f7f360ca6682b3c3f589de575c3
image: `perconalab/percona-server-mongodb-operator:PR-599-9dabe4f3`

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| self-healing | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| users | passed |
| upgrade-sharded | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/66af0afd45253f0299ce96165342b5863a64daf7
image: `perconalab/percona-server-mongodb-operator:PR-599-66af0afd`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| arbiter | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| users | passed |
| upgrade-sharded | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/b7e090c8e4eb012356aa7cc88d96a4d6a79ab32a
image: `perconalab/percona-server-mongodb-operator:PR-599-b7e090c8`

File: pkg/apis/psmdb/v1/psmdb_types.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (blez) [reply pkg/apis/psmdb/v1/psmdb_types.go]: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/40d27b8f01052a6979bf0fad7b5a2b89aed73133

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/37909fe419789dfe490021999353a008f6cb2502
image: `perconalab/percona-server-mongodb-operator:PR-599-37909fe4`

File: pkg/apis/psmdb/v1/psmdb_types.go [hunk 8]
```diff
@@ -549,12 +549,17 @@ func (cr *PerconaServerMongoDB) CompareVersion(version string) int {
 	return cr.Version().Compare(v.Must(v.NewVersion(version)))
 }
 
-func UserSecretNameInternal(cr *PerconaServerMongoDB) string {
-	internalPrefix := "internal-"
+const internalPrefix = "internal-"
+const userPostfix = "-users"
 
+func InternalSecretName(cr *PerconaServerMongoDB) string {
```
Comment (delgod) [inline pkg/apis/psmdb/v1/psmdb_types.go:555]:
```suggestion
func InternalUserSecretName(cr *PerconaServerMongoDB) string {
```
not clear it is ssl or users internal secret

**Review by delgod — CHANGES_REQUESTED** [review]

**Review by delgod — APPROVED** [review]

File: pkg/apis/psmdb/v1/psmdb_types.go (hunk 8 above)
Comment (delgod) [reply pkg/apis/psmdb/v1/psmdb_types.go]: fixed in https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/f8b55dda111811ca6860520dfc43ead2e16b4618

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/40d27b8f01052a6979bf0fad7b5a2b89aed73133
image: `perconalab/percona-server-mongodb-operator:PR-599-40d27b8f`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/ad391bd761ded43676fc41266aa2c1441c4c1bc8
image: `perconalab/percona-server-mongodb-operator:PR-599-ad391bd7`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/f8b55dda111811ca6860520dfc43ead2e16b4618
image: `perconalab/percona-server-mongodb-operator:PR-599-f8b55dda`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| operator-self-healing | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| arbiter | passed |
| service-per-pod | passed |
| upgrade | passed |
| liveness | passed |
| upgrade-sharded | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/599/commits/7cc96ebd8657a08c807a2ec3bd56f32ca218ff88
image: `perconalab/percona-server-mongodb-operator:PR-599-7cc96ebd`

**Pull request merged**